### DISCUSSION TITLE
Defining a a logging structure for a Python/.NET application

### DISCUSSION THREAD

**Discussion by kaihendry** [open-telemetry/community#778] [Community Discussions]
I'm asked to suggest a library for enforcing structured logging, especially with a Request/Trace ID.

The more I hear about Opentelemetry, the more I wonder, is there an interoperable structured logging format/structure that I could leverage in the project to achieve the above goals?

Thank you in advance,

**Comment by SergeyKanzhelev** [comment]
logs proto is not fully structured https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/logs/v1/logs.proto, but has some elements of a structure. You can read some discussions here: https://github.com/open-telemetry/opentelemetry-specification/issues/1752

As per libraries - in .NET ILogger is used, while Request is enforced by other library that exists in parallel - `Activity`. You can find articles on how the best use ILogger. I don't think you can enforce structure beyond attaching context from other library though. I don't know about python.

CC: @open-telemetry/dotnet-instrumentation-approvers @open-telemetry/python-approvers @open-telemetry/specs-logs-approvers

**Reply by kaihendry** [reply]
Is there a good place to ask about Python logging libraries targeting OpenTelemetry ?

**Reply by srikanthccv** [reply]
@kaihendry for python there is a library that inject trace context info by modifying the global log record factory https://opentelemetry-python-contrib.readthedocs.io/en/latest/instrumentation/logging/logging.html. And there is also an ongoing effort for initial version of logging sdk https://github.com/open-telemetry/opentelemetry-python/issues/1890 based on this OTEP https://github.com/open-telemetry/oteps/blob/main/text/logs/0150-logging-library-sdk.md.

**Reply by SergeyKanzhelev** [reply]
> Is there a good place to ask about Python logging libraries targeting OpenTelemetry ?

You can also ask here directly: https://github.com/open-telemetry/opentelemetry-python/discussions
